File: node/zkgroup/NotarySignature.ts
//
// Copyright 2020-2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

import ByteArray from './internal/ByteArray';

export default class NotarySignature extends ByteArray {
  static SIZE = 64;

  constructor(contents: Buffer) {
    super(contents, NotarySignature.checkLength(NotarySignature.SIZE));
  }
}

File: java/java/src/main/java/org/signal/zkgroup/NotarySignature.java
//
// Copyright 2020-2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

package org.signal.zkgroup;

import org.signal.zkgroup.internal.ByteArray;

public final class NotarySignature extends ByteArray {

  public static final int SIZE = 64;

  public NotarySignature(byte[] contents) throws InvalidInputException {
    super(contents, SIZE);
  }

}
